### PR TITLE
vale: install documentation and example styles

### PR DESCRIPTION
**Pull request by KAction**
"vale" is style checker for natural language. Currently, derivation only
installs binary, without any documentation on how to use it.  To
actually have "vale" to generate suggestions, config file with
specification of location of style files must be written.

This changeset adds "doc" output, which contains instructions on writing
config file and "data" output which contains style files.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/vale/default.nix [hunk 1]
```diff
@@ -1,10 +1,11 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
```
Comment (marsam) [inline pkgs/tools/text/vale/default.nix]: please revert this change, `buildGoPackage` is preferred when upstream vendorizes its dependencies

**Review by marsam — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/vale/default.nix (hunk 1 above)
Comment (KAction) [reply pkgs/tools/text/vale/default.nix]: Does it mean that documentation is inaccurate? This [section of Nixpkgs manual](https://nixos.org/nixpkgs/manual/#ssec-go-legacy) says exactly opposite:

    The function buildGoPackage builds legacy Go programs, not supporting Go modules.

Comment (marsam) [reply pkgs/tools/text/vale/default.nix:3]: This is debated, `buildGoPackage` is recommended when upstream vendorizes its dependencies (`vendor/` directory). See: https://github.com/NixOS/nixpkgs/issues/84826

**Comment by marsam** [comment]
Thank you, and sorry for the delay
